<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head>
<title>src/javax/xml/soap/SOAPException.java</title>
<meta name="robots" content="noindex,nofollow">
<meta name="generator" content="GLOBAL-6.6.3">
<meta http-equiv="Content-Style-Type" content="text/css">
<link rel="stylesheet" type="text/css" href="../style.css">
</head>
<body>
<a id="TOP" name="TOP"></a><div class="comment">
/* [&lt;][&gt;]<a href="#L104">[^]</a><a href="#L152">[v]</a>[top]<a href="#BOTTOM">[bottom]</a><a href="../mains.html">[index]</a><a href="../help.html">[help]</a> */</div>
<hr>
<h2 class="header">DEFINITIONS</h2>
This source file includes following definitions.
<ol>
<li><a href="#L104" title="Defined at 104.">getMessage</a></li>
<li><a href="#L122" title="Defined at 122.">getCause</a></li>
<li><a href="#L152" title="Defined at 152.">initCause</a></li>
</ol>
<hr>
<pre>
<div class="comment">
  Copyright (c) 2004, 2012, Oracle and/or its affiliates. All rights reserved.
  ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.</div>
<a id="L25" name="L25"></a>  25 
<a id="L26" name="L26"></a>  26 <strong class="reserved">package</strong> <a href="../D/26637.html" title="Multiple defined in 6 places.">javax</a>.xml.soap;
<a id="L27" name="L27"></a>  27 
<div class="comment">
  An exception that signals that a SOAP exception has occurred. A
  <code>SOAPException</code> object may contain a <code>String</code>
  that gives the reason for the exception, an embedded
  <code>Throwable</code> object, or both. This class provides methods
  for retrieving reason messages and for retrieving the embedded
  <code>Throwable</code> object.
  <p> Typical reasons for throwing a <code>SOAPException</code>
  object are problems such as difficulty setting a header, not being
  able to send a message, and not being able to get a connection with
  the provider.  Reasons for embedding a <code>Throwable</code>
  object include problems such as input/output errors or a parsing
  problem, such as an error in parsing a header.</p></div>
<a id="L43" name="L43"></a>  43 <strong class="reserved">public</strong> <strong class="reserved">class</strong> <a href="../R/6853.html" title="Multiple referred from 115 places.">SOAPException</a> <strong class="reserved">extends</strong> <a href="../S/1582.html#L45" title="Defined at 45 in src/java/lang/Exception.java.">Exception</a> <em class="brace">{</em>
<a id="L44" name="L44"></a>  44     <strong class="reserved">private</strong> <a href="../S/1726.html#L114" title="Defined at 114 in src/java/lang/Throwable.java.">Throwable</a> cause;
<a id="L45" name="L45"></a>  45 
<div class="comment">
      Constructs a <code>SOAPException</code> object with no
      reason or embedded <code>Throwable</code> object.</div>
<a id="L50" name="L50"></a>  50     <strong class="reserved">public</strong> SOAPException() <em class="brace">{</em>
<a id="L51" name="L51"></a>  51         <strong class="reserved">super</strong>();
<a id="L52" name="L52"></a>  52         <strong class="reserved">this</strong>.cause = <strong class="reserved">null</strong>;
<a id="L53" name="L53"></a>  53     <em class="brace">}</em>
<a id="L54" name="L54"></a>  54 
<div class="comment">
      Constructs a <code>SOAPException</code> object with the given
      <code>String</code> as the reason for the exception being thrown.
      @param reason a description of what caused the exception</div>
<a id="L61" name="L61"></a>  61     <strong class="reserved">public</strong> SOAPException(<a href="../D/8419.html" title="Multiple defined in 2 places.">String</a> reason) <em class="brace">{</em>
<a id="L62" name="L62"></a>  62         <strong class="reserved">super</strong>(reason);
<a id="L63" name="L63"></a>  63         <strong class="reserved">this</strong>.cause = <strong class="reserved">null</strong>;
<a id="L64" name="L64"></a>  64     <em class="brace">}</em>
<a id="L65" name="L65"></a>  65 
<div class="comment">
      Constructs a <code>SOAPException</code> object with the given
      <code>String</code> as the reason for the exception being thrown
      and the given <code>Throwable</code> object as an embedded
      exception.
      @param reason a description of what caused the exception
      @param cause a <code>Throwable</code> object that is to
             be embedded in this <code>SOAPException</code> object</div>
<a id="L76" name="L76"></a>  76     <strong class="reserved">public</strong> SOAPException(<a href="../D/8419.html" title="Multiple defined in 2 places.">String</a> reason, <a href="../S/1726.html#L114" title="Defined at 114 in src/java/lang/Throwable.java.">Throwable</a> cause) <em class="brace">{</em>
<a id="L77" name="L77"></a>  77         <strong class="reserved">super</strong>(reason);
<a id="L78" name="L78"></a>  78         <a href="../D/24507.html" title="Multiple defined in 7 places.">initCause</a>(cause);
<a id="L79" name="L79"></a>  79     <em class="brace">}</em>
<a id="L80" name="L80"></a>  80 
<div class="comment">
      Constructs a <code>SOAPException</code> object initialized
      with the given <code>Throwable</code> object.</div>
<a id="L85" name="L85"></a>  85     <strong class="reserved">public</strong> SOAPException(<a href="../S/1726.html#L114" title="Defined at 114 in src/java/lang/Throwable.java.">Throwable</a> cause) <em class="brace">{</em>
<a id="L86" name="L86"></a>  86         <strong class="reserved">super</strong>(cause.<a href="../D/35493.html" title="Multiple defined in 1021 places.">toString</a>());
<a id="L87" name="L87"></a>  87         <a href="../D/24507.html" title="Multiple defined in 7 places.">initCause</a>(cause);
<a id="L88" name="L88"></a>  88     <em class="brace">}</em>
<a id="L89" name="L89"></a>  89 
<div class="comment">
      Returns the detail message for this <code>SOAPException</code>
      object.
      <p>
      If there is an embedded <code>Throwable</code> object, and if the
      <code>SOAPException</code> object has no detail message of its
      own, this method will return the detail message from the embedded
      <code>Throwable</code> object.
      @return the error or warning message for this
              <code>SOAPException</code> or, if it has none, the
              message of the embedded <code>Throwable</code> object,
              if there is one</p></div>
<a id="L104" name="L104"></a> 104     <strong class="reserved">public</strong> <a href="../D/8419.html" title="Multiple defined in 2 places.">String</a> <a href="../R/17668.html" title="Multiple referred from 607 places.">getMessage</a>() <em class="brace">{</em>
<a id="L105" name="L105"></a> 105         <a href="../D/8419.html" title="Multiple defined in 2 places.">String</a> <a href="../D/27673.html" title="Multiple defined in 6 places.">message</a> = <strong class="reserved">super</strong>.<a href="../D/20565.html" title="Multiple defined in 56 places.">getMessage</a>();
<a id="L106" name="L106"></a> 106         <strong class="reserved">if</strong> (<a href="../D/27673.html" title="Multiple defined in 6 places.">message</a> == <strong class="reserved">null</strong> &amp;&amp; cause != <strong class="reserved">null</strong>) <em class="brace">{</em>
<a id="L107" name="L107"></a> 107             <strong class="reserved">return</strong> cause.<a href="../D/20565.html" title="Multiple defined in 56 places.">getMessage</a>();
<a id="L108" name="L108"></a> 108         <em class="brace">}</em> <strong class="reserved">else</strong> <em class="brace">{</em>
<a id="L109" name="L109"></a> 109             <strong class="reserved">return</strong> <a href="../D/27673.html" title="Multiple defined in 6 places.">message</a>;
<a id="L110" name="L110"></a> 110         <em class="brace">}</em>
<a id="L111" name="L111"></a> 111     <em class="brace">}</em>
<a id="L112" name="L112"></a> 112 
<div class="comment">
      Returns the <code>Throwable</code> object embedded in this
      <code>SOAPException</code> if there is one. Otherwise, this method
      returns <code>null</code>.
      @return the embedded <code>Throwable</code> object or <code>null</code>
              if there is none</div>
<a id="L121" name="L121"></a> 121 
<a id="L122" name="L122"></a> 122     <strong class="reserved">public</strong> <a href="../S/1726.html#L114" title="Defined at 114 in src/java/lang/Throwable.java.">Throwable</a> <a href="../R/15467.html" title="Multiple referred from 93 places.">getCause</a>() <em class="brace">{</em>
<a id="L123" name="L123"></a> 123         <strong class="reserved">return</strong> cause;
<a id="L124" name="L124"></a> 124     <em class="brace">}</em>
<a id="L125" name="L125"></a> 125 
<div class="comment">
      Initializes the <code>cause</code> field of this <code>SOAPException</code>
      object with the given <code>Throwable</code> object.
      <p>
      This method can be called at most once.  It is generally called from
      within the constructor or immediately after the constructor has
      returned a new <code>SOAPException</code> object.
      If this <code>SOAPException</code> object was created with the
      constructor {@link #SOAPException(Throwable)} or
      {@link #SOAPException(String,Throwable)}, meaning that its
      <code>cause</code> field already has a value, this method cannot be
      called even once.
      @param  cause the <code>Throwable</code> object that caused this
              <code>SOAPException</code> object to be thrown.  The value of this
              parameter is saved for later retrieval by the
              {@link #getCause()} method.  A <tt>null</tt> value is
              permitted and indicates that the cause is nonexistent or
              unknown.
      @return  a reference to this <code>SOAPException</code> instance
      @throws IllegalArgumentException if <code>cause</code> is this
              <code>Throwable</code> object.  (A <code>Throwable</code> object
              cannot be its own cause.)
      @throws IllegalStateException if the cause for this <code>SOAPException</code> object
              has already been initialized</p></div>
<a id="L152" name="L152"></a> 152     <strong class="reserved">public</strong> <strong class="reserved">synchronized</strong> <a href="../S/1726.html#L114" title="Defined at 114 in src/java/lang/Throwable.java.">Throwable</a> <a href="../R/20547.html" title="Multiple referred from 873 places.">initCause</a>(<a href="../S/1726.html#L114" title="Defined at 114 in src/java/lang/Throwable.java.">Throwable</a> cause) <em class="brace">{</em>
<a id="L153" name="L153"></a> 153         <strong class="reserved">if</strong> (<strong class="reserved">this</strong>.cause != <strong class="reserved">null</strong>) <em class="brace">{</em>
<a id="L154" name="L154"></a> 154             <strong class="reserved">throw</strong> <strong class="reserved">new</strong> <a href="../S/1555.html#L38" title="Defined at 38 in src/java/lang/IllegalStateException.java.">IllegalStateException</a>("Can't override cause");
<a id="L155" name="L155"></a> 155         <em class="brace">}</em>
<a id="L156" name="L156"></a> 156         <strong class="reserved">if</strong> (cause == <strong class="reserved">this</strong>) <em class="brace">{</em>
<a id="L157" name="L157"></a> 157             <strong class="reserved">throw</strong> <strong class="reserved">new</strong> <a href="../D/4055.html" title="Multiple defined in 2 places.">IllegalArgumentException</a>("Self-causation not permitted");
<a id="L158" name="L158"></a> 158         <em class="brace">}</em>
<a id="L159" name="L159"></a> 159         <strong class="reserved">this</strong>.cause = cause;
<a id="L160" name="L160"></a> 160 
<a id="L161" name="L161"></a> 161         <strong class="reserved">return</strong> <strong class="reserved">this</strong>;
<a id="L162" name="L162"></a> 162     <em class="brace">}</em>
<a id="L163" name="L163"></a> 163 <em class="brace">}</em>
</pre>
<hr>
<div class="comment">
/* [&lt;][&gt;]<a href="#L104">[^]</a><a href="#L152">[v]</a><a href="#TOP">[top]</a>[bottom]<a href="../mains.html">[index]</a><a href="../help.html">[help]</a> */</div>
</body></html>